cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -1728,6 +1728,37 @@ class AzureFileSystem::Impl {
     }
     return Status::OK();
   }
+
+  Status DeleteFile(const AzureLocation& location) {
+    if (location.path.empty()) {
+      return Status::Invalid("Cannot delete an empty path");
+    }
+
+    auto last_sep = location.path.find_last_of(internal::kSep);
+    auto directory = location.path.substr(0, last_sep);
+    auto file = location.path.substr(last_sep + 1);
+
+    auto file_client =
+      datalake_service_client_->GetFileSystemClient(location.container)
+        .GetDirectoryClient(directory).GetFileClient(file);
```

Comment:
You can `GetFileClient(location.path)` from filesystem client -- you don't have to create a directory client.